internal/manager/config_validation.go

```diff
@@ -0,0 +1,29 @@
+package manager
+
+import (
+	"errors"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// This file contains a set of constructors that are used to validate and set validated values in Config.
+// They're meant to be used together with ValidatedValue[T] type.
+
+func createNamespacedName(s string) (types.NamespacedName, error) {
```

Comment:
Nit: I'm not sure if naming those functions `<Type>FromFlagValue` or something like this , wouldn't be better.